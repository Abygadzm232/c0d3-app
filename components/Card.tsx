import React from 'react'
import { ReactComponent as Checked } from '../assets/images/checked.svg'
import { ReactComponent as Fail } from '../public/curriculumAssets/icons/exclamation.svg'

type Props = {
  children?: React.ReactNode
  type?: 'success' | 'fail'
  text?: string
  classes?: string
  title: string
}

const iconProps: { [type: string]: string } = {
  className: 'mb-4',
  width: '100px',
  height: '100px'
}

const icons: { [type: string]: React.FC } = {
  success: Checked,
  fail: Fail
}

const Card: React.FC<Props> = props => {
  const classes = `${props.classes ||
    'col-sm-8 col-md-7 col-lg-6 col-xl-6 m-auto px-md-5 border-0'}`
  const Icon = props.type ? icons[props.type] : null
  return (
    <div className="row mt-5">
      <div className={`card shadow-sm ${classes}`}>
        <div className="card-body text-center pt-5 pb-5">
          {Icon && <Icon {...iconProps} />}
          <h1 className="card-title h2 font-weight-bold mb-5">{props.title}</h1>
          <p className="card-text">{props.text}</p>
          {props.children}
        </div>
      </div>
    </div>
  )
}

export default Card
