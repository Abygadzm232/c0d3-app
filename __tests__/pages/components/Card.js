import React from 'react'
import renderer from 'react-test-renderer'
import Card from '../../../components/Card'

describe('Snapshot for Card', () => {
  test('Card', () => {
    const component = renderer.create(<Card title="Title of the card" />)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('Card with Text', () => {
    const component = renderer.create(
      <Card
        title="Title of the card"
        text="Neque porro quisquam est qui dolorem ipsum quia dolor sit amet, consectetur, adipisci velit..."
      />
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('Card with Childrens', () => {
    const component = renderer.create(
      <Card
        title="Title of the card"
        text="Neque porro quisquam est qui dolorem ipsum quia dolor sit amet, consectetur, adipisci velit..."
      >
        <div className="input-group mb-3">
          <div className="input-group-prepend">
            <span className="input-group-text" id="inputGroup-sizing-default">
              Default
            </span>
          </div>
          <input
            type="text"
            className="form-control"
            aria-label="Sizing example input"
            aria-describedby="inputGroup-sizing-default"
          />
          <div className="col-5">
            <button type="button" className="btn btn-dark">
              Primary
            </button>
          </div>
        </div>
      </Card>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('Card with Success', () => {
    const component = renderer.create(
      <Card
        success
        title="Title of the card"
        text="Neque porro quisquam est qui dolorem ipsum quia dolor sit amet, consectetur, adipisci velit..."
      />
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
