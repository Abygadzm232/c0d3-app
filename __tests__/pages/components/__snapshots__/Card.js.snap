// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot for Card Card 1`] = `
<div
  className="row mt-5"
>
  <div
    className="card shadow-sm col-sm-8 col-md-7 col-lg-6 col-xl-4 m-auto"
  >
    <div
      className="card-body text-center pt-5 pb-5"
    >
      <h1
        className="card-title h2 font-weight-bold"
      >
        Title of the card
      </h1>
      <p
        className="card-text"
      />
    </div>
  </div>
</div>
`;

exports[`Snapshot for Card Card with Childrens 1`] = `
<div
  className="row mt-5"
>
  <div
    className="card shadow-sm col-sm-8 col-md-7 col-lg-6 col-xl-4 m-auto"
  >
    <div
      className="card-body text-center pt-5 pb-5"
    >
      <h1
        className="card-title h2 font-weight-bold"
      >
        Title of the card
      </h1>
      <p
        className="card-text"
      >
        Neque porro quisquam est qui dolorem ipsum quia dolor sit amet, consectetur, adipisci velit...
      </p>
      <div
        className="input-group mb-3"
      >
        <div
          className="input-group-prepend"
        >
          <span
            className="input-group-text"
            id="inputGroup-sizing-default"
          >
            Default
          </span>
        </div>
        <input
          aria-describedby="inputGroup-sizing-default"
          aria-label="Sizing example input"
          className="form-control"
          type="text"
        />
        <div
          className="col-5"
        >
          <button
            className="btn btn-dark"
            type="button"
          >
            Primary
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Snapshot for Card Card with Success 1`] = `
<div
  className="row mt-5"
>
  <div
    className="card shadow-sm col-sm-8 col-md-7 col-lg-6 col-xl-4 m-auto"
  >
    <div
      className="card-body text-center pt-5 pb-5"
    >
      <svg
        className="mb-4"
        height="100px"
        viewBox="0 0 426.667 426.667"
        width="100px"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          'data-old_color'="#000000"
          'data-original'="#000000"
          className="active-path"
          d="M293.333 135.04L190.08 240.213l-52.907-53.12-28.373 28.374 83.413 83.2 133.974-129.92z"
          fill="#00CF92"
        />
        <path
          'data-old_color'="#000000"
          'data-original'="#000000"
          className="active-path"
          d="M213.333 0C95.513 0 0 95.513 0 213.333s95.513 213.333 213.333 213.333 213.333-95.513 213.333-213.333S331.154 0 213.333 0zm0 388.053c-96.495 0-174.72-78.225-174.72-174.72s78.225-174.72 174.72-174.72c96.446.117 174.602 78.273 174.72 174.72 0 96.496-78.224 174.72-174.72 174.72z"
          fill="#00CF92"
        />
      </svg>
      <h1
        className="card-title h2 font-weight-bold"
      >
        Title of the card
      </h1>
      <p
        className="card-text"
      >
        Neque porro quisquam est qui dolorem ipsum quia dolor sit amet, consectetur, adipisci velit...
      </p>
    </div>
  </div>
</div>
`;

exports[`Snapshot for Card Card with Text 1`] = `
<div
  className="row mt-5"
>
  <div
    className="card shadow-sm col-sm-8 col-md-7 col-lg-6 col-xl-4 m-auto"
  >
    <div
      className="card-body text-center pt-5 pb-5"
    >
      <h1
        className="card-title h2 font-weight-bold"
      >
        Title of the card
      </h1>
      <p
        className="card-text"
      >
        Neque porro quisquam est qui dolorem ipsum quia dolor sit amet, consectetur, adipisci velit...
      </p>
    </div>
  </div>
</div>
`;
